File: backend/app/schemas/coverage.py
from pydantic import BaseModel


class CoverageRow(BaseModel):
    key: str
    label: str
    type: str
    frecuencia_anual: int
    esperadas: int
    realizadas: int
    estado: str


class CoverageMarkIn(BaseModel):
    theme_key: str
    covered: bool

File: frontend/src/lib/coverage.ts
import api from "@/lib/api"

export type CoverageEstado = "en_tiempo" | "riesgo" | "atrasado" | "critico"

export interface CoverageRow {
  key: string
  label: string
  type: string
  frecuencia_anual: number
  esperadas: number
  realizadas: number
  estado: CoverageEstado
}

export async function getCobertura(): Promise<CoverageRow[]> {
  const r = await api.get<CoverageRow[]>("/annual-plan/cobertura")
  return r.data
}
